File: src/test/java/koorkah/TestAny.java
package koorkah;

import org.junit.jupiter.api.Test;

class TestAny extends BaseT
{

    @Test
    void testAny()
    {
        String expression = koorkah.toRegEx("`a` any{`b`} `c`");
        assertRegexIs("ab*c", expression);
        assertRegexMatchesText(expression, "abc");
        assertRegexMatchesText(expression, "abbbc");
        assertRegexMatchesText(expression, "ac");
        assertRegexDoesntMatchText(expression, "asc");       
    }

}

File: src/python/test/TestAny.py


import unittest

import Utils
import Refi

class TestAny(unittest.TestCase):

    def setUp(self):
        self.refi = Refi.Refi()
        self.utils = Utils.Utils();

    def test_Any(self):
     
        expression = self.refi.toRegEx("`a` any{`b`} `c`")
        self.utils.assertRegexIs("ab*c", expression);
        self.utils.assertRegexMatchesText(expression, "abc");
        self.utils.assertRegexMatchesText(expression, "abbbc");
        self.utils.assertRegexMatchesText(expression, "ac");
        self.utils.assertRegexDoesntMatchText(expression, "asc"); 

    def runTest(self):
        pass  

if __name__ == '__main__':
    unittest.main()